vm/option.go

```diff
@@ -50,44 +54,72 @@ func NewOption[T any](namespace string, defaultConfig T, optionFunc OptionFunc[T
 	return newOptionWithBytes(namespace, configOptionFunc)
 }
 
-func WithBuilder() RegisterFunc {
-	return func(vm *VM) {
-		vm.builder = builder.NewManual(vm)
-	}
+func WithBuilder() Opt {
+	return newFuncOption(func(o *Options) {
+		o.builder = true
+	})
 }
 
-func WithGossiper() RegisterFunc {
-	return func(vm *VM) {
-		vm.gossiper = gossiper.NewManual(vm)
-	}
+func WithGossiper() Opt {
+	return newFuncOption(func(o *Options) {
+		o.gossiper = true
+	})
 }
 
 func WithManual() Option {
 	return NewOption[struct{}](
 		"manual",
 		struct{}{},
-		func(vm *VM, _ struct{}) error {
-			WithBuilder()(vm)
-			WithGossiper()(vm)
-			return nil
+		func(_ api.VM, _ struct{}) (Opt, error) {
+			return newFuncOption(func(o *Options) {
+				WithBuilder().apply(o)
+				WithGossiper().apply(o)
+			}), nil
```

Comment:
(not necessary for this PR) - we should change this to a single exported option `WithManual` there's no need to export the others individually when they can only be used currently to set the gossiper/builder to the manual implementation.

We should also separate the API / subscription related options from this package, so that the API implementations themselves do not depend on the VM package and the VM package can import these options. This way you can use the API / subscription related options without importing the VM package (which should not be necessary to use them)